### ISSUE TITLE
Remove BaseAggregationTestCase in favor of AbstractSerializingTestCase

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#42894]
Many aggregation builders use [`BaseAggregationTestCase`](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/search/aggregations/BaseAggregationTestCase.java) (which extends [`AbstractBuilderTestCase`](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java)) to check xcontent serialization, shuffling, equals/hashcode, etc.  

These are basically very specialized, agg-specific forms of the newer [`AbstractSerializingTestCase`](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java) unique to the agg framework.  It would be nice if we could get rid of this specialized test class and standardize on `AbstractSerializingTestCase`.

I implemented [`WeightedAvgAggregationBuilderTests`](https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/search/aggregations/metrics/weighted_avg/WeightedAvgAggregationBuilderTests.java) recently using `AbstractSerializingTestCase` and everything went smoothly, so I think converting the rest over should be doable.  But `BaseAggregationTestCase` does contain a lot of specialized logic, so we need to check and ensure that all the existing tests can be replicated by `AbstractSerializingTestCase` (or perhaps a sub-class that includes the needed functionality)

**Comment by wchaparro** [comment]
@icemakegirl yes please, it is still relevant. Thanks!  feel free to ask if you need any help.

**Comment by icemakegirl** [comment]
@wchaparro The BaseAgregationTestCase file contains all the Agg framework tests and do I have to pass them to another file or are there other types of tests in it?
